pkg/operator/ceph/object/dependents.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright 2021 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package object
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	v1 "github.com/rook/rook/pkg/apis/ceph.rook.io/v1"
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/daemon/ceph/client"
+	"github.com/rook/rook/pkg/util/dependents"
+)
+
+// CephObjectStoreDependents returns the buckets which exist in the object store that should block
+// deletion.
+func CephObjectStoreDependents(
+	clusterdCtx *clusterd.Context,
+	clusterInfo *client.ClusterInfo,
+	store *v1.CephObjectStore,
+) (*dependents.DependentList, error) {
+	nsName := fmt.Sprintf("%s/%s", store.Namespace, store.Name)
+	baseErrMsg := fmt.Sprintf("failed to get dependents of CephObjectStore %q", nsName)
+
+	deps := dependents.NewDependentList()
+
+	objCtx, err := NewMultisiteContext(clusterdCtx, clusterInfo, store)
+	if err != nil {
+		return deps, errors.Wrapf(err, "%s. failed to get context", baseErrMsg)
+	}
+	opsCtx, err := NewMultisiteAdminOpsContext(objCtx, &store.Spec)
+	if err != nil {
+		return deps, errors.Wrapf(err, "%s. failed to get admin API context", baseErrMsg)
+	}
+
+	buckets, err := opsCtx.AdminOpsClient.ListBuckets(context.TODO())
+	if err != nil {
+		return deps, errors.Wrapf(err, "%s. failed to list buckets in CephObjectStore %q", baseErrMsg, nsName)
+	}
+	foundCheckerBucket := false
+	for _, b := range buckets {
+		if strings.HasPrefix(b, s3HealthCheckBucketName+"-") && !foundCheckerBucket {
+			// Rook will only ever create a single s3 health checker bucket
+			foundCheckerBucket = true
```

Comment:
Sorry, I don't fully understand the `foundcheckerBucket` variable, say if somehow if there are two buckets `s3HealthCheckBucketName` with the prefix, we only ignore the first one and others will be added to the list. But how will the user know which is actually bucket linked with `healthchecker`.
Consider there are two buckets `rook-ceph-bucket-checker-abc` and `rook-ceph-bucket-checker-xyz`(linked to health checker) if the below skips `rook-ceph-bucket-checker-abc` and include `rook-ceph-bucket-checker-xyz` into the dependency list, then we are providing wrong info for the user right? I am not sure when the health checker deletes actually delete the bucket or not sure what happens if the pod get restarted

Reply:
That's a good point.

I think I can adjust this since we know what the health checker uses as its bucket name.